src/main/kotlin/com/example/delivery/mapper/OrderMapper.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.delivery.mapper
+
+import com.example.delivery.domain.DomainOrder
+import com.example.delivery.dto.request.CreateOrderDTO
+import com.example.delivery.dto.response.OrderDTO
+import com.example.delivery.dto.response.ShipmentDetailsDTO
+import com.example.delivery.mongo.MongoOrder
+import org.bson.types.ObjectId
+import java.math.BigDecimal
+
+object OrderMapper {
+    fun CreateOrderDTO.toMongo(): MongoOrder = MongoOrder(
+        items = items.mapKeys { ObjectId(it.key) },
+        shipmentDetails = shipmentDetails.toModel(),
+        userId = ObjectId(userId),
+    )
+
+    fun DomainOrder.toDTO(): OrderDTO = OrderDTO(
+        (id ?: "none").toString(),
+        items.mapKeys { it.key.toString() },
+        totalPrice,
+        shipmentDetails.toDTO(),
+        status,
+        (id ?: "none").toString(),
+    )
+
+    fun DomainOrder.toMongo(): MongoOrder = MongoOrder(
+        id, items, totalPrice, shipmentDetails, MongoOrder.Status.valueOf(status)
+    )
+
+    fun MongoOrder.toDomain(): DomainOrder = DomainOrder(
+        id,
+        items = items ?: mutableMapOf(),
```

Comment:
```suggestion
        items = items ?: emptyMap(),
```
prefer immutable collections whenever it's possible